libs/basetests/langchain_basetests/unit_tests.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC, abstractmethod
+from typing import Type
+
+import pytest
+from langchain_core.language_models import BaseChatModel
+from langchain_core.pydantic_v1 import BaseModel, Field
+from langchain_core.tools import tool
+
+
+class Person(BaseModel):
+    name: str = Field(..., description="The name of the person.")
+    age: int = Field(..., description="The age of the person.")
+
+
+@tool
+def my_adder_tool(a: int, b: int) -> int:
+    """Takes two integers, a and b, and returns their sum."""
+    return a + b
+
+
+class BaseUnitTests(ABC):
```

Comment:
Could we rename this so that it's obvious it's for chat models?

Reply:
I was originally thinking of having just one class for everything - but ya makes sense to split by type